pkg/tsdb/cloudwatch/services/list_metrics.go

```diff
@@ -26,23 +28,15 @@ func (*ListMetricsService) GetHardCodedDimensionKeysByNamespace(namespace string
 	return dimensionKeys, nil
 }
 
-func (l *ListMetricsService) GetDimensionKeysByDimensionFilter(r *models.DimensionKeysRequest) ([]string, error) {
+func (l *ListMetricsService) GetDimensionKeysByDimensionFilter(r *request.DimensionKeysRequest) ([]string, error) {
 	input := &cloudwatch.ListMetricsInput{}
 	if r.Namespace != "" {
 		input.Namespace = aws.String(r.Namespace)
 	}
 	if r.MetricName != "" {
 		input.MetricName = aws.String(r.MetricName)
 	}
-	for _, dimension := range r.DimensionFilter {
-		df := &cloudwatch.DimensionFilter{
-			Name: aws.String(dimension.Name),
-		}
-		if dimension.Value != "" {
-			df.Value = aws.String(dimension.Value)
-		}
-		input.Dimensions = append(input.Dimensions, df)
-	}
+	setDimensionFilter(input, r.DimensionFilter)
```

Comment:
Moving this piece of code to a utility since it's reused by the dimension values method.